aws-xray-recorder-sdk-core/src/test/java/com/amazonaws/xray/entities/EntityTest.java

```diff
@@ -76,6 +76,18 @@ void testUnknownClassSerialization() {
         seg.serialize();  // Verify we don't crash here
     }
 
+    @Test
+    void testPrecursorIds() {
+        Segment seg = new SegmentImpl(AWSXRay.getGlobalRecorder(), "test");
+        SubsegmentImpl subseg = new SubsegmentImpl(AWSXRay.getGlobalRecorder(), "test", seg);
+        subseg.addPrecursorId("myId1");
+        subseg.addPrecursorId("myId2");
+        String serializedSubSeg = subseg.serialize();
+
+        String expected = "\"precursor_ids\":[\"myId1\",\"myId2\"]";
+        assertThat(serializedSubSeg).contains(expected);
+    }
+
```

Comment:
Is there any way to mock the behaviour of adding precursor IDs in 2 threads and ensure that the new data structure handles that situation correctly?